syntax = "proto3";

package ShuHai.gRPC.Examples.Streaming;

service ProcessUtil
{
    rpc GetProcesses(GetProcessesRequest) returns(stream ProcessInfo) {}
    rpc PostProcesses(stream ProcessInfo) returns(PostProcessesReply) {}
    rpc MatchProcesses(stream ProcessInfo) returns(stream ProcessInfo) {}
}

message GetProcessesRequest
{
    string filter = 1;
}

message PostProcessesReply
{
    bool ok = 1;
}

message ProcessInfo
{
    uint32 pid = 1;
    string name = 2;
}
